redaxo/src/core/lib/api_function.php

```diff
@@ -81,6 +81,16 @@ public static function factory()
 
         $api = rex_request(self::REQ_CALL_PARAM, 'string');
 
+        if (empty($api)) {
+            $messageBody = file_get_contents('php://input');
+            if (!empty($messageBody)) {
+                parse_str($messageBody, $data);
+                if (isset($data[self::REQ_CALL_PARAM]) && is_string($data[self::REQ_CALL_PARAM])) {
+                    $api = rex_type::cast($data[self::REQ_CALL_PARAM], 'string');
```

Comment:
Hier ist mir die Verwendung noch nicht so klar. Man setzt dann einen PUT oder DELETE-Request o.ä. ab, gestaltet den Body aber wie bei POST-Requests? Also dort stecken dann die Api-Parameter im Query-Format drin.

Das kommt mir spontan komisch vor. Aber vielleicht erstmal: Hab ich es so richtig kapiert?

Reply:
Im Body könnten verschiedene Daten enhalten sein, somit ist diese Änderung vom Prinzip her nicht vollständig.
PHP führt leider keine automatische Erkennung durch, sodass keine Abfrage per $_POST oder $_GET möglich ist und man die Auswertung des Bodys selbst bewerkstelligen muss.
Hier könnte ich noch um eine Abfrage des Content-Types/Media Types ergänzen, sodass "application/x-www-form-urlencoded", "application/json" oder "multipart/form-data" erkannt und angemessen ausgewertet werden können.

Würde die Erweiterung der Einfachheit halber erstmal bei  "application/x-www-form-urlencoded" und "application/json" belassen.
Wenn Bedarf besteht, würde ich noch um "multipart/form-data" erweitern.

Ist das so in Ordnung?
Ich kann diese Änderung auch in einem speraten Pull-Request unterbringen, wenn gewünscht.

Reply:
Ich finde komisch, dass hier der Api-Param aus dem Request-Body geholt wird.
Wäre es nicht schöner, bei PUT/DELETE den api-param mit in die URL zu packen, sodass man im Body eben beliebige Daten packen kann. Eben auch JSON etc.
So müsste es doch auch aktuell schon nutzbar sein.

Reply:
OK, ?rex-api-call= in der URL funktioniert. Die Änderung ist damit hinfällig.